spyder/widgets/browser.py

```diff
@@ -52,28 +84,154 @@ def acceptNavigationRequest(self, url, navigation_type, isMainFrame):
             url, navigation_type, isMainFrame)
 
 
-class WebView(QWebEngineView):
-    """Web view"""
+class WebView(QWebEngineView, SpyderWidgetMixin):
+    """
+    Web view.
+    """
+
     def __init__(self, parent, handle_links=True):
-        QWebEngineView.__init__(self, parent)
+        super().__init__(parent)
+
         self.zoom_factor = 1.
-        self.zoom_out_action = create_action(self, _("Zoom out"),
-                                             icon=ima.icon('zoom_out'),
-                                             triggered=self.zoom_out)
-        self.zoom_in_action = create_action(self, _("Zoom in"),
-                                            icon=ima.icon('zoom_in'),
-                                            triggered=self.zoom_in)
+        self.context_menu = None
+
         if WEBENGINE:
             if handle_links:
                 web_page = WebPage(self)
             else:
                 web_page = QWebEnginePage(self)
+
             self.setPage(web_page)
             self.source_text = ''
 
+    def setup(self, options={}):
+        # Actions
+        original_back_action = self.pageAction(QWebEnginePage.Back)
+        back_action = self.create_action(
+            name=WebViewActions.Back,
+            text=_("Back"),
+            icon=self.create_icon('previous'),
+            triggered=lambda: original_back_action.trigger(),
+            context=Qt.WidgetWithChildrenShortcut,
+        )
+        original_forward_action = self.pageAction(QWebEnginePage.Forward)

```

Comment:
Please add a blank line above this one to follow the same convention you established in the lines below.

Reply:
Fixed